ui/editor/src/view.js

```diff
@@ -157,7 +163,20 @@ function sparePieces(ctrl, color, orientation, position) {
     return m('div', {
         class: containerClass,
         onmousedown: function() {
-          ctrl.vm.selected(piece);
+          if (['pointer', 'trash'].indexOf(piece) !== -1) {
+            ctrl.vm.selected(piece);
+          } else {
+            var listener;
+            ctrl.vm.draggingSpare(true);
+            ctrl.vm.selected('pointer');
+
+            document.addEventListener('mouseup', listener = function() {
+              ctrl.vm.selected(piece);
+              ctrl.vm.draggingSpare(false);
+              m.redraw();
+              document.removeEventListener('mouseup', listener);
+            });
```

Comment:
I didn't feel good about it, but it worked pretty well for me to my surprise. Any suggestions to make it less hacky?